pkg/jobs/mem_jobs_test.go

```diff
@@ -184,6 +184,33 @@ func TestRetry(t *testing.T) {
 	w.Wait()
 }
 
+func TestPanic(t *testing.T) {
+	var w sync.WaitGroup
+
+	maxExecCount := 4
+
+	broker := NewMemBroker("panic", WorkersList{
+		"panic": {
+			Concurrency:  1,
+			MaxExecCount: uint(maxExecCount),
+			RetryDelay:   1 * time.Millisecond,
+			WorkerFunc: func(ctx context.Context, _ *Message) error {
+				w.Done()
+				panic("oops")
+			},
+		},
+	})
+
+	w.Add(maxExecCount)
+	_, _, err := broker.PushJob(&JobRequest{
+		WorkerType: "panic",
+		Message:    nil,
+	})
+
+	assert.NoError(t, err)
+	w.Wait()
```

Comment:
This test checks that we the job that has paniced is retried, which is what a really wanted to test.

I could indeed add a test checking that the worker is still available even after a panic.